### DISCUSSION TITLE
Is there anyway to slice a plane or line in MultiFab?

### DISCUSSION THREAD

**Discussion by jiecaix** [AMReX-Codes/amrex#3662] [Q&A] [answered]
For example, I want to represent the electric field with MultiFab, assuming its range is (0-100e-6, 0-100e-6, 0-100e-6), then I want to output the electric field in the plane of x = 50e-6, is there anyway in Amrex to convert it to a plane data?

**Comment by drummerdoc** [comment]
Since MultiFabs have no concept of grid spacing and physical coordinates, the short answer is no.  If you do the arithmetic to find the integer of the plane you want, you can make a box with the same centering as the MultiFab, and do a copy-on-intersect.

**Reply by WeiqunZhang** [reply, marked as answer]
There is a function that may help, but probably does not do exactly what you want because it only works on cell centered data.

https://amrex-codes.github.io/amrex/doxygen/namespaceamrex.html#a3a431197f206b148529308a7796d5039

However, it should not be hard to extend it to other index types.

There is also a function called `get_line_data`

https://amrex-codes.github.io/amrex/doxygen/namespaceamrex.html#a7f08bece47186514a8d7ec979e4b87ca

One can use it to extract a line of data.

Note that both functions return a MultiFab, which is a distributed data container. That is the data might be on multiple processes.

**Reply by jiecaix** [reply]
Thank you guys for help, thats helpful for me.
And i have a further question, if i get two plane by ```amrex::get_slice_data```, is there any function in Amrex could directly interplate by position i give with 2 known plane point.

**Reply by WeiqunZhang** [reply]
In a general case, the two slices may have different Boxes (in the 2D) and they may be distributed differently among MPI processes. So we have to consider that. Moreover, on fine AMR levels, they may not cover the same 2D domain. Assuming the domain issue is not a problem, what you could do is something like below.
```
const int normal_direction = ...;
auto slice1 = get_slice_data(normal_direction, ....);
auto slice2 = get_slice-data(normal_direction,...);
int ndiff = slice2->boxArray()[0][normal_direction] - slice1->boxArray()[0][normal_direction];
BoxArray ba = slice1->boxArray();
ba.shift(normal_direction, ndiff);
MultiFab other_slice2(ba, slice1->DistributionMapping(), slice1->nComp(), 0);
other_slice2.ParallelCopy(slice2);

// now it's safe to operate on slice1 and other_slice2
for (MFIter mfi(*slice1); mfi.isValid(); ++mfi) {
    auto const& fab1 = (*slice1)[mfi];
    auto const& fab2 = other_slice2[mfi];
}
```

**Reply by jiecaix** [reply]
Thanks, that's great!
